cw_mib_generator.py

```diff
@@ -0,0 +1,94 @@
+# @file cwc_mib_generator.py
+#
+# Copyright (C) 2016  Metaswitch Networks Ltd
+
+import re
+import os.path
+
+COMMON_MIB = "./CLEARWATER-MIB-COMMON"
+PC_EXTRAS = "./CLEARWATER-MIB-PC-EXTRAS"
+CWC_EXTRAS = "../CLEARWATER-MIB-CWC-EXTRAS"
+
+PC_MIB_NAME = "./PROJECT-CLEARWATER-MIB"
+CWC_MIB_NAME = "../METASWITCH-CLEARWATER-CORE-MIB"
+
+
+def find_header(data, header_regex):
+    header_index = None
+    for i, line in enumerate(data):
+        if header_regex.match(line):
+            header_index = i
+            break
+
+    return header_index
+
+
+def find_end_statement(data):
+    end_regex = re.compile(r"^END")
+    return find_header(data, end_regex)
+
+
+def find_section(data, section_regex):
+    generic_header_regex = re.compile(r"(^\-\- [\w ]+|^END)")
+
+    module_start_index = find_header(data, section_regex)
+    module_end_index = module_start_index + \
+                       find_header(data[module_start_index+1:], 
+                                   generic_header_regex)
+
+    return module_start_index, module_end_index
+
+
+def append_compliance(data, compliance_objects):
+    compliance_regex = re.compile(r"\-\- Module Compliance")
+    compliance_section = find_section(data, compliance_regex)
+    data[compliance_section[1]:compliance_section[1]] = compliance_objects[1:]
```

Comment:
I don't really like the use of magic numbers, here.  To understand this, I have to work out that `find_section()` is returning line indices for the start and end of the section, and know something about why it is necessary to strip out the first line of the `compliance_objects` list.

I think it would be clearer if `find_section()` returned a simple `Section` object, that contained `start_index` and `end_index` members.